### ISSUE TITLE
(de)serialization

### ISSUE THREAD

**Issue by bjansen** [bjansen/gyokuro#3]
The current `jsonSerializer` was good enough for a PoC, but I think it'd be better if we provided an extension point like we did for template engines:

``` ceylon
shared interface Serializer {
    shared formal String serialize(Object o);
    shared formal Type deserialize<Type>(String serialized);
    shared formal [String+] contentTypes;
}
```

And then provide implementations based on:
- the serialization feature introduced in 1.2 (https://github.com/tombentley/alabama ?)
- Jackson
- Gson
- other non-JSON serializers?

The correct serializer would be picked based on the request's content type.

Also provide a deserializer that reads the body of a request and creates an object that can be passed to handlers:

``` ceylon
void handler(serialized MyDomainObject obj) { }
```

`serialized` indicates that a deserializer should be used to create the instance. The correct deserializer would be picked based on the request's content type.

**Comment by renatoathaydes** [comment]
In my opinion this interface absolutely belongs in a serialization API, not a web framework. Doesn't the Ceylon SDK already provide something like this??

**Comment by bjansen** [comment]
I suppose I could use this: https://modules.ceylon-lang.org/repo/1/ceylon/language/1.2.0/module-doc/api/serialization/index.html

But unfortunately I have no way to specify which content types are supported by each serializer.

**Comment by bjansen** [comment]
Fixed in cce51480880deec9c66f0c717c737c6fe4852a92 and 5eee1f7b289fd3c9482fdc3c1942fb846e867b8e.
